src/ovirtapi/transform.js

```diff
@@ -486,6 +486,7 @@ const Snapshot = {
   toApi ({ snapshot }: { snapshot: SnapshotType }): ApiSnapshotType {
     return {
       description: snapshot.description,
+      persist_memorystate: snapshot.persistMemoryState.toString(),
```

Comment:
```suggestion
      persist_memorystate: convertBool(snapshot.persistMemoryState),
```

Also, the `persist_memorystate` type in `ApiSnapshotType` should be a boolean not a string as per add POST: http://ovirt.github.io/ovirt-engine-api-model/master/#services/snapshots/methods/add and the Snapshot struct: http://ovirt.github.io/ovirt-engine-api-model/master/#types/snapshot

Reply:
@sjd78 according to this [link](http://ovirt.github.io/ovirt-engine-api-model/master/#types/snapshot):
it should be a [boolean](http://ovirt.github.io/ovirt-engine-api-model/master/#types/boolean): so it might be "false" OR "true".
https://github.com/oVirt/ovirt-web-ui/blob/2886959acd3588132d5fc276a9d4df043de19644/src/ovirtapi/types.js#L50
I can change this type to: `persist_memorystate: "true" | "false"`.
But if I understood you correctly it shouldn't be changed more than explicitly mention the possible values, if I didn't can you please explain what did you meant?
Regarding the suggested change it converts it the opposite direction (string to bool).

Reply:
Hmm, and what about the sentence _The valid values are the strings `false` and `true`._
in http://ovirt.github.io/ovirt-engine-api-model/master/#types/boolean ?
@sjd78 I am little bit confused 😄